src/vs/platform/update/electron-main/updateService.darwin.ts

```diff
@@ -47,16 +49,16 @@ export class DarwinUpdateService extends AbstractUpdateService {
 		this.setState(State.Idle);
 	}
 
-	protected setUpdateFeedUrl(quality: string): boolean {
+	protected buildUpdateFeedUrl(quality: string): string | undefined {
+		const url = createUpdateURL('darwin', quality);
 		try {
-			electron.autoUpdater.setFeedURL(createUpdateURL('darwin', quality));
+			electron.autoUpdater.setFeedURL(url);
```

Comment:
Not too happy that `buildUpdateFeedUrl` causes state to change in this service... but oh well

Reply:
not nice, but no bad side effects?

Reply:
On a second thought, it shouldn't behave different than before. The method is called as before, it just returns a string instead of a boolean... The name should maybe indicate the state-fullness but oh well